File: OneDrive/Escritorio/Proyecto SP2/Proyecto/creditos/CreditosApi/Models/Usuario.cs
using System.Collections.Generic;

namespace CreditosApi.Models
{
    public class Usuario
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = string.Empty;
        public string Apellido { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Dpi { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public DateTime FechaRegistro { get; set; } = DateTime.Now;

        // 🔹 Relación 1:N (un usuario tiene muchos clientes)
        public ICollection<Cliente> Clientes { get; set; } = new List<Cliente>();
         public ICollection<Item> Items { get; set; } = new List<Item>();
    }
}

File: OneDrive/Escritorio/Proyecto SP2/Proyecto/creditos/lib/models/usuario.dart
class Usuario {
  final int id;
  final String nombre;
  final String apellido;
  final String email;
  final String dpi;
  final String fechaRegistro;

  Usuario({
    required this.id,
    required this.nombre,
    required this.apellido,
    required this.email,
    required this.dpi,
    required this.fechaRegistro,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'],
      nombre: json['nombre'],
      apellido: json['apellido'],
      email: json['email'],
      dpi: json['dpi'], 
      fechaRegistro: json['fechaRegistro'],
    );
  }
}
